src/core/lib/surface/channel.h

```diff
@@ -110,8 +110,9 @@ struct grpc_channel {
   grpc_core::ManualConstructor<grpc_core::CallRegistrationTable>
       registration_table;
   grpc_core::RefCountedPtr<grpc_core::channelz::ChannelNode> channelz_node;
+  grpc_core::ManualConstructor<grpc_core::MemoryAllocator> allocator;
```

Comment:
Instead of adding more `ManualConstructor` data members, how about we go ahead and change `grpc_channel` to be allocated via `new` instead of `gpr_malloc()`, as per the TODO above?

Reply:
That will be a bigger change: grpc_channel right now is allocated inline with the filter stack.
Let's punt on that until we're doing the filter stack changes.

Reply:
We don't actually need to change the way it's allocated; we can use placement-new instead.

But I also can live with punting on this for now.